### ISSUE TITLE
[Bug]: Critical: mailer transports main is ignored

### ISSUE THREAD

**Issue by Cruiser13** [pimcore/pimcore#15824]
### Pimcore version

v11.0.7, upgraded from Pimcore 10 with same settings

### Steps to reproduce

Add a custom smtp into config.yaml like this:
```yaml
framework:
#### SYMFONY MAILER TRANSPORTS
    mailer:
        transports:
            main: 'smtp://mail@host.com:pass@smtp.domain.com:587'
```

Confirm that the framework var is being set and used  using
**php bin/console debug:config framework**
and
**php bin/console debug:config --resolve-env framework**

### Actual Behavior

Pimcore does use native mail sending - directly from the server it is on.

### Expected Behavior

Send mail by smtp server as defined

**Comment by gadget60** [comment]
Just tested in PC 11.0.7 and could not reproduce this error: The transport is being used as defined via "framework.mailer.transports.main".
Is your transport URI literal as shown above? If yes, mind the notice about special character encoding mentioned on https://symfony.com/doc/current/mailer.html

**Comment by dvesh3** [comment]
works for me as well:
![image](https://github.com/pimcore/pimcore/assets/5137917/8aa35f4b-d6f3-4349-ab1d-264c795db1e7)

**Comment by Cruiser13** [comment]
@dvesh3 I think you misunderstood me here. If I run **php bin/console debug:config --resolve-env framework mailer** it will properly give me the output **main: 'smtp://mail@host.com:pass@smtp.domain.com:587'** as expected. But if I send any mail (even using the send test mail function in the pimcore admin) it'll use the native sending instead of the smtp one.
I'm happy to provide a test environment if needed, we have multiple instances showing this behaviour.
If we provide the same SMTP details directly in the **Symfony\Component\Mailer\Transport\Dsn** class it'll properly send via smtp.

It looks like pimcore does ignore the setting somehow.

**Comment by dvesh3** [comment]
Hi @Cruiser13 we need some more information (app env, exact transport used, ) to debug this issue as it seems to works for us even when sending emails from UI. We simply leverage mailer from Syfmony so i would also check bugs on Symfony for the version used. I don't know if it helps to look at the test environment without access to the logs.

**Comment by Cruiser13** [comment]
Hi @dvesh3 thank you for your reply.
We run pimcore in a LAMP stack (Managed Server by hetzner) with PHP 8.1 and Apache 2.4. Transport used is main.
Symfony Mailer is on version 6.3.5. We did setup a clean symfony flex project using this mailer version and have not been able to reproduce the issue there. We checked the symfony issues on github before opening this issue and again now. It looks like pimcore does not properly initialize vendor/symfony/mailer/Transport/Dsn.php class.

This issue appeared when upgrading pimcore 10 to 11, no changes to the config.yml have been done.

I'm happy to provide full access to the test environment including ssh, logs or anything else needed to track this down. Also open for a video call to demonstrate the issue directly - find my contact info on https://pringuin.de/kontakt

**Comment by MarcoGuiducci** [comment]
Hi,

I have the same problem reported by @Cruiser13 on version v.11.1.1.
Basically, I have configured my SMTP server and filled the "main" transport in the config.yml file.
I've tried to send a "test email" by the user interface: Pimcore says that the email is correctly sent but no email arrived effectively.

Personally, I've configured the "pimcore_newsletter" transport exactly as the "main" one, and the email started correctly.
This fixed my problem, but I'm not sure that this is the correct approach

**Comment by Cruiser13** [comment]
@robertSt7 any chance that somebody else can look into this?
I can confirm @MarcoGuiducci here - if I configure "pimcore_newsletter" transport too, it'll use smtp correctly.

**Comment by brusch** [comment]
Don't see any issue here, just tested it locally using Mailhog as well as with AWS SES using the following config: 

```yaml
framework:
    mailer:
        transports:
            main: smtp://mailhog:1025?verify_peer=0
            pimcore_newsletter: smtp://mailhog:1025?verify_peer=0
```

Used latest Pimcore version.

**Comment by Cruiser13** [comment]
@brusch you need to remove pimcore_newsletter - main should be enough according to the docs but apparently pimcore does use pimcore_newsletter as default instead of main as it did on pimcore X.

**Comment by brusch** [comment]
@Cruiser13 thanks for pointing out, my fault ... so this issue actually is that Symfony Mailer is not by default using the `main` transport, BUT it uses the first transport in the config, see https://symfony.com/doc/6.3/mailer.html#multiple-email-transports 😉 

The question now is whether we should change the behavior here or not, i mean we could, but I'm not sure if this is a good idea 🤔

**Comment by brusch** [comment]
Actually I think we could fix it, because the fix would only affect `\Pimcore\Mail\Mailer` and not the behavior when using Symfony Mailer directly. Preparing a patch 👍

**Comment by brusch** [comment]
Fix created: https://github.com/pimcore/ee-pimcore/pull/18

**Comment by Cruiser13** [comment]
@brusch I can not see the fix (repository is private I guess). I'm not sure if the issue is coming from the first transport in the config - we only defined the main transport in our config (as written in my first comment) and experienced the native sending from our controllers and from admin.

**Comment by brusch** [comment]
@Cruiser13 yes, it's coming from that, because when the newsletter bundle is enabled, then it defines the transport before the core-bundle 😉 

Private repo: yes, that's intentional, we'll explain the reason behind this change soon 👍

**Comment by dpfaffenbauer** [comment]
@brusch will pimcore community edition be a downstream project for enterprise?

**Comment by brusch** [comment]
@dpfaffenbauer sorry for the late answer: **NO** pimcore/pimcore will remain the main project - we just did some tests here.

**Comment by jahwood** [comment]
Same Problem in Version: v11.1.2 , fixed with added pimcore_newsletter to the config with same DSN

**Comment by kingjia90** [comment]
it's officially fixed in 11.1.4 , please upgrade to any version equal or above it
